rskj-core/src/main/java/co/rsk/core/bc/PendingStateImpl.java

```diff
@@ -75,49 +76,52 @@
     private Repository pendingStateRepository;
     private final TxPendingValidator validator = new TxPendingValidator();
 
-    public PendingStateImpl(Blockchain blockChain,
-                            BlockStore blockStore,
+    public PendingStateImpl(BlockStore blockStore,
+                            ReceiptStore receiptStore,
                             EthereumListener listener,
                             ProgramInvokeFactory programInvokeFactory,
                             Repository repository,
                             RskSystemProperties config) {
         this(config,
-                blockChain,
                 repository,
                 blockStore,
+                receiptStore,
                 programInvokeFactory,
                 listener,
                 config.txOutdatedThreshold(),
                 config.txOutdatedTimeout());
     }
 
     public PendingStateImpl(RskSystemProperties config,
-                            Blockchain blockChain,
                             Repository repository,
                             BlockStore blockStore,
+                            ReceiptStore receiptStore,
                             ProgramInvokeFactory programInvokeFactory,
                             EthereumListener listener,
                             int outdatedThreshold,
                             int outdatedTimeout) {
         this.config = config;
-        this.blockChain = blockChain;
         this.blockStore = blockStore;
         this.repository = repository;
+        this.receiptStore = receiptStore;
         this.programInvokeFactory = programInvokeFactory;
         this.listener = listener;
         this.outdatedThreshold = outdatedThreshold;
         this.outdatedTimeout = outdatedTimeout;
 
         this.pendingStateRepository = repository.startTracking();
-        this.bestBlock = blockChain.getBestBlock();
 
         if (this.outdatedTimeout > 0) {
             this.cleanerTimer = Executors.newSingleThreadScheduledExecutor(r -> new Thread(r, "PendingStateCleanerTimer"));
         }
     }
 
     @Override
-    public void start() {
+    public void start(Block initialBestBlock) {
+        if (bestBlock == null) {
```

Comment:
Why don't accept initialBestBlock directly? use case to invoke start method, having bestBlock not null?

Reply:
As we spoke offline, I removed this